File: src/main/java/com/example/demo/Entity/HubMaster.java
package com.example.demo.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "hub_master")
@Data
public class HubMaster {

	
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "hub_id")
    private int hubId;
	
	
	@Column(name = "hub_name")
	private String hubName;
	
	@Column(name = "hub_address_and_details", columnDefinition = "TEXT")
	private String hubAddressAndDetails;
	    
	@Column(name = "contact_number", unique = true) 
	private Long contactNumber;
	    
	@ManyToOne
	@JoinColumn(name = "city_id")
	private CityMaster city;
	    
	@ManyToOne
	@JoinColumn(name = "state_id")
	private StateMaster state;

	public int getHubId() {
		return hubId;
	}

	public void setHubId(int hubId) {
		this.hubId = hubId;
	}

	public String getHubName() {
		return hubName;
	}

	public void setHubName(String hubName) {
		this.hubName = hubName;
	}

	public String getHubAddressAndDetails() {
		return hubAddressAndDetails;
	}

	public void setHubAddressAndDetails(String hubAddressAndDetails) {
		this.hubAddressAndDetails = hubAddressAndDetails;
	}

	public Long getContactNumber() {
		return contactNumber;
	}

	public void setContactNumber(Long contactNumber) {
		this.contactNumber = contactNumber;
	}

	public CityMaster getCity() {
		return city;
	}

	public void setCity(CityMaster city) {
		this.city = city;
	}

	public StateMaster getState() {
		return state;
	}

	public void setState(StateMaster state) {
		this.state = state;
	}

	 
}

File: Backend.NET/Models/HubMaster.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FleetManagementSystem.Api.Models;

[Table("hub_master")]
public class HubMaster
{
    [Key]
    [Column("hub_id")]
    public int HubId { get; set; }

    [Column("hub_name")]
    public string? HubName { get; set; }

    [Column("hub_address_and_details", TypeName = "TEXT")]
    public string? HubAddressAndDetails { get; set; }

    [Column("contact_number")]
    public long ContactNumber { get; set; }

    [ForeignKey("City")]
    [Column("city_id")]
    public int? CityId { get; set; }
    public CityMaster? City { get; set; }

    [ForeignKey("State")]
    [Column("state_id")]
    public int? StateId { get; set; }
    public StateMaster? State { get; set; }
}
